source/extensions/filters/common/rbac/utility.h

```diff
@@ -12,6 +14,13 @@ namespace Filters {
 namespace Common {
 namespace RBAC {
 
+class DynamicMetadataKeys {
+public:
+  const std::string ShadowPolicyIdField{"shadow_effective_policyID"};
```

Comment:
Does this key and the other keys in this file need to be documented somewhere? Is the idea here that this is meant to be sent in logging data? If we have well known metadata keys I think they should be documented somewhere?

Reply:
It may be useful to document these constants as other filters may be able to augment Envoy's behavior based on these. I think that should be done along with the rename suggested in [this comment](https://github.com/envoyproxy/envoy/pull/4910#pullrequestreview-171255352) in another PR.

Reply:
OK if you plan on doing a follow up, SGTM.

Reply:
Will do. Let me know if you have suggestions on where to document these.

Reply:
Tracked via #4981.